Source/Scene/TimeDynamicPointCloud.js

```diff
@@ -681,6 +711,21 @@ define([
             loadFrame(this, nextInterval, updateState, frameState);
         }
 
+        var that = this;
+        if (defined(frame) && !defined(this._lastRenderedFrame)) {
+            frameState.afterRender.push(function() {
+                that._readyPromise.resolve(that);
+            });
+        }
+
+        if (defined(frame) && (frame !== this._lastRenderedFrame)) {
+            if (that.frameChanged.numberOfListeners > 0) {
+                frameState.afterRender.push(function() {
+                    that.frameChanged.raiseEvent(that);
```

Comment:
Works well, however, in requestRender mode, the frames stop advancing and this event stops firing after each frame has been loaded once. I think this may be an argument for handling the frame advancement in a clock listener rather than in the render loop.

Reply:
I understand your point about wanting to keep it simple, but on the other hand, I think separating the update and render logic also has value.

The biggest concern I have is since the update is tied to the rendering, that event won't trigger when in request render mode, so there's no way to watch and explicitly render a new frame when needed. I can't attest to the use cases for this as much as you, but it seems like a good candidate for requestRender mode in that you only need to render a frame (assuming nothing else is changing) when the timeline advances enough. However if you think it won't get used, then this doesn't need to be a priority now.

Reply:
I think its best to revisit later if it comes up. The render and update are just too linked together at the moment, sometimes by necessity, and it might take some time to get the separation right.